packages/block-editor/src/components/block-list/index.native.js

```diff
@@ -59,12 +59,17 @@ export class BlockList extends Component {
 	}
 
 	renderDefaultBlockAppender() {
+		const { shouldShowInsertionPointBefore } = this.props;
+		const willShowInsertionPoint = shouldShowInsertionPointBefore(); // call without the client_id argument since this is the appender
 		return (
 			<ReadableContentView>
-				<BlockListAppender
-					rootClientId={ this.props.rootClientId }
-					renderAppender={ this.props.renderAppender }
-				/>
+				{ willShowInsertionPoint && this.renderAddBlockSeparator() }
+				{ ! willShowInsertionPoint &&
+					<BlockListAppender
+						rootClientId={ this.props.rootClientId }
+						renderAppender={ this.props.renderAppender }
+					/>
+				}
```

Comment:
Just a personal preference, but I find a ternary in situations like this easier to quickly read than having both `{ aBoolean && something }` and `{ ! aBoolean && somethingElse }` because it makes the branching nature of the logic more explicit. In other words, something like:

```
{ willShowInsertionPoint
	? this.renderAddBlockSeparator()
	: <BlockListAppender
		rootClientId={ this.props.rootClientId }
		renderAppender={ this.props.renderAppender }
	  />
}
```
Feel free to leave this as-is if you prefer.

Reply:
Makes sense, addressed with 16565b0. Notice, the line wrapping syntax is slightly different than your suggestion to conform with the linter.